aces-nifi-processors/src/main/java/net/acesinc/nifi/processors/mongodb/PartialUpdateMongo.java

```diff
@@ -307,26 +311,83 @@ protected void transferFlowFile(FlowFile f, Object result, ProcessContext contex
 
                 if (OPERATION_CURRENT_DATE.equals(operation)) {
                     operationValues.append(p, true);
-                } else if (OPERATION_ADD_TO_SET.equals(operation) && doc.get(p) != null) {
-                    String jsonString = mapper.writeValueAsString(doc.get(p));
-                    //looking for an array to use $each mode
-                    if (jsonString.contains("[")) {
-                        List<Map<String, Object>> operationValueDocs = mapper.readValue(jsonString, List.class);
+                } else if (OPERATION_ADD_TO_SET.equals(operation) && (doc.get(p) != null || p.contains("."))) {
+                    if (!p.contains(".")) {
+                        String jsonString = mapper.writeValueAsString(doc.get(p));
+                        //looking for an array to use $each mode
+                        if (jsonString.contains("[{")) {
+                            //logic for when there is an array of objects in $addToSet-$each mode
+                            logger.debug("jsonString [ " + jsonString + " ].");
```

Comment:
Do these debug statements need to stay?

Reply:
1) debug statements are all turned off in the default logback.xml of nifi, so we will not see these w/o great effort for starters
2) LOC 320:  We need to keep 320.  That is where an array of objects has been detected. I can elaborate on that debug statement more.
3) LOC 332:  We can remove this particular debug as that is regular array $addToSet, the bread and basic butter scenario
4) All other debug statements w/ parent.child being analyzed need to remain.
5) Any thoughts on LOC 689, 793?  warn vs error level.  Currently, these are warn. Saw this situation as more of a warn at the time and trying to give a more quiet complaint, but still give one and have things work when this scenario is used and it's not an array.